### PR TITLE
Announcement_infost flag indicates sparring; identify unit_action_data_attack flag

### PR DESCRIPTION
**Pull request by Bumber64**
#714

Accidentally deleted repository before merge.

### CODE REVIEW

File: df.units.xml [hunk 1]
```diff
@@ -2324,22 +2324,22 @@
...
+            <flag-bit name='heavy' comment='(HEAVY)'/>
+            <flag-bit name='wild' comment='(WILD)'/>
+            <flag-bit name='precise' comment='(PRECISE)'/>
+            <flag-bit name='charge' comment='(INDEPENDENT_MULTIATTACK)'/>
+            <flag-bit name='bad_multiattack' comment='(BAD_MULTIATTACK)'/>
+            <flag-bit name='zero_criticals' comment='(ZERO_CRITICALS)'/>
+            <flag-bit name='lightly_tap' comment='(NO_LETHAL_INTENT)'/>
+            <flag-bit name='automatic_hit' comment='(AUTOMATIC_HIT)'/>
+            <flag-bit name='sparring_hit' comment='(SPARRING_HIT)'/>
+            <flag-bit name='spar_report' comment='()'/>
```
Comment (Bumber64) [inline df.units.xml:2342]: Should `charge` be renamed `independent_multiattack`?
What's the Bay12 name for `spar_report`?

**Comment by Bumber64** [comment]
Coordinate with DFHack/dfhack#4247

File: df.units.xml (hunk 1 above)
Comment (myk002) [reply df.units.xml:2342]: I'd say keep `charge` as it is.

There does not appear to be a bit for `spar_report` -- the header just goes up to SPARRING_HIT. That flag bit may need to be removed

Comment (ab9rf) [reply df.units.xml:2342]: i can't find any indication for a use of bit15 and believe we can safely remove `spar_report`

Comment (Bumber64) [reply df.units.xml]: > I'd say keep `charge` as it is.

Does it correctly describe the behavior? I.e., is the bit set during a charge?

**Comment by Bumber64** [comment]
If you've got a list of all the `unit_move_*` to replace `unit_action_data_*`, we can handle that in this PR.

File: df.units.xml (hunk 1 above)
Comment (myk002) [reply df.units.xml]: We just have a general policy of "if it's named, then don't change it unless it's egregious". Behavior notes can go in the comment.

**Comment by myk002** [comment]
Do you mean
```cpp
#define UNIT_MOVE_MOVE_FLAG_CHARGE BIT1
#define UNIT_MOVE_MOVE_FLAG_FROM_SPARRING BIT2
```
?

**Comment by myk002** [comment]
actually, let's get this one merged so we can resolve the build error. Further changes can be in a separate PR

**Review by myk002 — APPROVED** [review]

**Pull request merged**